presto-orc/src/main/java/com/facebook/presto/orc/DwrfStreamOrderingConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.orc.proto.DwrfProto.KeyInfo;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class DwrfStreamOrderingConfig
+{
+    // column is the logical columns index for a table with n columns
+    // This index range from 0 to n - 1
+    private final ImmutableMap<Integer, List<KeyInfo>> columnToKeyList;
+    private final ImmutableMap<Integer, Set<KeyInfo>> columnToKeySet;
```

Comment:
Do you keep `columnToKeyList` just for the sake of the order of KeyInfos? ImmutableSet in columnToKeySet preserves iteration order the same as when the elements have been added to the set.

It's one of the features of Guava's immutable collections https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableCollection.html:
```
Deterministic iteration. The iteration order is always well-defined, depending on how the collection 
was created (see the appropriate factory method for details). View collections such as [Multiset.elementSet()]
(https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Multiset.html#elementSet()) iterate in 
the same order as the parent, except as noted.
```

Reply:
Yes. I didn't know that ImmutableSet preserves input order. Will collapse them into one